### DISCUSSION TITLE
Stockfish to play like only tactical players or only positional players?

### DISCUSSION THREAD

**Discussion by HarshilPatel007** [official-stockfish/Stockfish#5708] [Q&A]
Hi all,
Is there any way to set up , make changes to stockfish so that it can play only or more like positional or tactical game?

How does the chess program modify the engine to play like only tactical players or positional players? Because , I've seen this in a few chess programs that, their bots are specifically catered to one playing style so that we can play and train against some specific playing style. This is a very helpful way to train ourselves.

**Comment by marlonwq** [comment]
> Oi pessoalExiste alguma maneira de configurar , fazer alterações no stockfish para que ele possa jogar apenas ou mais como um jogo posicional ou tático?
> 
> Como o programa de xadrez modifica o motor para jogar apenas como jogadores táticos ou posicionais? Porque , eu vi isso em alguns programas de xadrez que, seus bots são especificamente atendidos a um estilo de jogo para que possamos jogar e treinar contra algum estilo de jogo específico. Esta é uma maneira muito útil de nos treinarmos.

“A tactical combination is the result of a good positional game”, of course in many positions more than one play is humanly good and possible.

Depending on the opponent's style he will choose one or the other, just look at the recent clashes between Ding x Gukesh, where in certain positions there were two types of spears, both good, and the human factor of "playing the riskiest" or "playing the calmer" he decided. 

However, the computer does not have this emotional side, it reproduces what is most accurate, based on hundreds of calculations. For example, a human can filter out some good moves and some bad moves by simply looking at a position and analyzing it for a few seconds, a computer calculates all the moves.

The human factor also weighs heavily in terms of play style with "positional" or "tactical" play, so it is currently impossible to make the computer behave in a certain way, precisely because it simply works on "raw calculation".

However, what you asked for could be an idea of ​​choice functionality based on the style of player x or y, like a “what would he play”. Would this style be defined by analyzing matches and in certain positions where there were different possibilities, seeing what he chose and what would be the possible variants from there, slower, more confusing? However, we have not yet evolved to that point.
